### PR TITLE
Use UI Request Flag

### PR DESCRIPTION
**Pull request by mwood23**
Closes # 

## 💸 TL;DR

This bumps devvit and adds a flag to enabling ui reuqest to the settings modal.

## 📜 Details
[Design Doc]()

[Jira]()

## 🧪 Testing Steps / Validation

https://github.com/reddit/play/assets/13633613/4293cfbc-3f16-44c6-8ebe-773b9a921363

## ✅ Checks

- [x] CI tests (if present) are passing
- [x] Adheres to code style for repo
- [x] Contributor License Agreement (CLA) completed if not a Reddit employee

### CODE REVIEW

File: src/elements/play-settings-dialog.ts [hunk 1]
```diff
@@ -237,6 +239,18 @@ export class PlaySettingsDialog extends LitElement {
             Use experimental blocks. Default:
             ${onOff(defaultSettings.useExperimentalBlocks)}.
           </label>
+          <label>
+            <input
+              ?checked="${this.useUIRequest}"
+              type="checkbox"
+              @change=${(ev: Event & {currentTarget: HTMLInputElement}) =>
+                this.dispatchEvent(
```
Comment (niedzielski) [inline src/elements/play-settings-dialog.ts:247]: Would you mind updating the HTMLElementEventMap at the top?

**Review by gunsch — COMMENTED** [review]
Code looks good!

I don't love the name "UI Request", I think it's pretty orthogonal to (my understanding of) what the KMP stuff does, and idk how settled we are on that through the codebase already. Is this the time to question it or are we past that point?

**Review by niedzielski — APPROVED** [review]
:+1: :+1:

**Comment by mwood23** [comment]
@gunsch It's not too late to change IMO! This is just what we've been calling it for a few months. I think multithreading is a more apt description potentially. I went with what I saw Android had to be consistent.

**Review by niedzielski — APPROVED** [review]

**Pull request merged**